src/app/profile/overview/connected-accounts/connected-accounts.component.ts

```diff
@@ -0,0 +1,71 @@
+import { Component, OnDestroy, OnInit, ViewEncapsulation } from '@angular/core';
+import { Subscription } from 'rxjs';
+
+import { Context, Contexts } from 'ngx-fabric8-wit';
+import { Observable } from 'rxjs/Observable';
+import { Space, Spaces, SpaceService } from 'ngx-fabric8-wit';
+import { AuthenticationService, UserService, User } from 'ngx-login-client';
+
+import { ProviderService } from '../../../getting-started/services/provider.service';
+import { Fabric8UIConfig } from '../shared/config/fabric8-ui-config';
+import { Http, Headers, RequestOptions, RequestOptionsArgs, Response } from '@angular/http';
+import { ExtUser, GettingStartedService } from '../../../getting-started/services/getting-started.service';
+
+@Component({
+  encapsulation: ViewEncapsulation.None,
+  selector: 'alm-connected-accounts',
+  templateUrl: './connected-accounts.component.html',
+  styleUrls: ['./connected-accounts.component.less'],
+  providers: [ GettingStartedService, ProviderService ]
+})
+export class ConnectedAccountsComponent implements OnDestroy, OnInit {
+  context: Context;
+  loggedInUser: User;
+  subscriptions: Subscription[] = [];
+  contextSubscription: Subscription;
+
+  authGitHub: boolean = false;
+  authOpenShift: boolean = false;
+  gitHubLinked: boolean = false;
+  openShiftLinked: boolean = false;
+
+  userName: string;
+  contextUserName: string;
+  githubUserName: string;
+
+  constructor(
+    private contexts: Contexts,
+    private auth: AuthenticationService,
+    private gettingStartedService: GettingStartedService,
+    private http: Http,
+    private providerService: ProviderService,
+    private userService: UserService) {
+    // for GitHub
+    this.subscriptions.push(auth.gitHubToken.subscribe(token => {
+      this.gitHubLinked = (token !== undefined && token.length !== 0);
+    }));
+    // for OpenShift
+    this.contextSubscription = this.contexts.current.subscribe(val => {
```

Comment:
You can use this.subscriptions.push(...) instead of this.contextSubscription.

Reply:
I believe that is the next line - would that mean that I only need one? And then how would I call the `{{contextUsername}}` to display the OpenShift username?

Reply:
You just don't need a contextSubscription variable. The code would look more like this:

this.subscriptions.push(this.contexts.current.subscribe(val => {
  this.contextUserName = val.user.attributes.username;
});

Reply:
should be all set now